### PR TITLE
Rebrand priority sampling while being BC with legacy approach to set it

### PR DESCRIPTION
**Pull request by labbati**
The purpose of this PR is to deprecate the `setSamplingPriority()` method on the span object and to provide a new API to force tracing sampling via a new tag `manual.keep` or `manual.drop`.

AC:
- Sampling priority metric and tag should be kept in sync
- There should be no difference in using the legacy method and the new tag based method

Note:
Based on other python leading implementation the business logic has been kept super simple. So tag setting is one shot and there is not way to come back once the user set the value. So the only API exposed is `.setTag(<name>,  true)`.

### CODE REVIEW

File: dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java [hunk 1]
```diff
@@ -24,6 +24,14 @@
 
   Integer getSamplingPriority();
 
+  /**
+   * @deprecated Use {@link MutableSpan#setTag(String, boolean)} instead using either tag names
```
Comment (tylerbenson) [inline dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java]: Could you suggest using OT's `Span` instead?

File: dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Class exposing tag names to mark a span to be dropped or kept by the sampler based on user
+ * intent.
+ */
+public final class ForcedTracing {
+
+  /** The user has decided to drop the trace. */
+  public static final String manual_DROP = "manual.drop";
+
+  /** The user has decided to keep the trace. */
+  public static final String manual_KEEP = "manual.keep";
```
Comment (tylerbenson) [inline dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java]: Static finals should be all uppercase, so `MANUAL_KEEP`.

File: dd-trace-ot/src/main/java/datadog/opentracing/DDSpanContext.java [hunk 3]
```diff
@@ -335,6 +352,25 @@ public synchronized void setTag(final String tag, final Object value) {
     }
   }
 
+  /**
+   * Sets priority sampling based on the forced tracing tag provided by the user.
+   *
+   * @param tag
+   * @param value
+   */
+  private void processForcedTracingTag(final String tag, Object value) {
```
Comment (tylerbenson) [inline dd-trace-ot/src/main/java/datadog/opentracing/DDSpanContext.java]: This should be implemented via a "decorator" instead.  See `AnalyticsSampleRateDecorator` for an example.

File: dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java [hunk 4]
```diff
@@ -24,6 +24,14 @@
...
+   *     {@link datadog.trace.api.sampling.ForcedTracing#manual_KEEP} or {@link
+   *     datadog.trace.api.sampling.ForcedTracing#manual_DROP}.
+   * @param newPriority
+   * @return
+   */
+  @Deprecated
   MutableSpan setSamplingPriority(final int newPriority);
```
Comment (tylerbenson) [inline dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java]: Most customers are probably not using this directly.  Perhaps the `PrioritySampling` class should also be deprecated?

File: dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package datadog.trace.api.sampling;
+
+/**
+ * Class exposing tag names to mark a span to be dropped or kept by the sampler based on user
+ * intent.
+ */
+public final class ForcedTracing {
```
Comment (tylerbenson) [inline dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java]: Should we consider putting these in `DDTags` instead?

File: dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java (hunk 1 above)
Comment (labbati) [reply dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java]: Good idea!

File: dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java (hunk 4 above)
Comment (labbati) [reply dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java]: Not sure about this. While customers do not use `PrioritySampling`, `PrioritySampling` is still a real thing for us. So  we should consider to un-deprecate `datadog.trace.common.sampling.PrioritySampling` is we want to deprecate  `datadog.trace.api.sampling.PrioritySampling`. WDYT?

File: dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java (hunk 5 above)
Comment (labbati) [reply dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java]: This is based on requirements, I think it is intended that this is `ForcedTracing`

File: dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java (hunk 2 above)
Comment (labbati) [reply dd-trace-api/src/main/java/datadog/trace/api/sampling/ForcedTracing.java]: Again, this is based on requirements, I think it is intended that this is mixed case. Will add more details here.

File: dd-trace-ot/src/main/java/datadog/opentracing/DDSpanContext.java (hunk 3 above)
Comment (labbati) [reply dd-trace-ot/src/main/java/datadog/opentracing/DDSpanContext.java]: As we discussed this in person, I will move this to a decorator and create a note to refactor the decorator workflow in order to improve and make the API more clear. Thanks for the useful conversation.

File: dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java (hunk 4 above)
Comment (tylerbenson) [reply dd-trace-api/src/main/java/datadog/trace/api/interceptor/MutableSpan.java]: oh, ok.  I thought this was potentially going away too.  If not, then never mind.

**Comment by labbati** [comment]
@tylerbenson I think that all your comments have been addresses, back to you for final code review

File: dd-trace-ot/src/main/java/datadog/opentracing/decorators/DDDecoratorsFactory.java [hunk 6]
```diff
@@ -21,6 +21,8 @@
         new SpanTypeDecorator(),
         new Status5XXDecorator(),
         new Status404Decorator(),
-        new URLAsResourceName());
+        new URLAsResourceName(),
+        new ForceManualKeepDecorator(),
+        new ForceManualDropDecorator());
```
Comment (tylerbenson) [inline dd-trace-ot/src/main/java/datadog/opentracing/decorators/DDDecoratorsFactory.java]: I think this list is alphabetical.  Please sort.

**Comment by labbati** [comment]
@tylerbenson addressed your final comment, could you please give the final approval 😄

**Pull request merged**